common/trtexec_vendor/CMakeLists.txt

```diff
@@ -0,0 +1,73 @@
+cmake_minimum_required(VERSION 3.14)
+project(trtexec_vendor)
+
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 17)
+  set(CMAKE_CXX_STANDARD_REQUIRED ON)
+  set(CMAKE_CXX_EXTENSIONS OFF)
+endif()
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  add_compile_options(-Wall -Wextra -Wpedantic)
+endif()
+
+find_package(ament_cmake REQUIRED)
+find_package(CUDA REQUIRED)
+find_package(cudnn_cmake_module REQUIRED)
+find_package(CUDNN REQUIRED)
+find_package(tensorrt_cmake_module REQUIRED)
+find_package(TENSORRT REQUIRED)
+
+if(NOT (${CUDA_FOUND} AND ${CUDNN_FOUND} AND ${TENSORRT_FOUND}))
+  message(WARNING "cuda, cudnn, tensorrt libraries are not found")
+  return()
+endif()
+
+set(TRTEXEC_DEFAULT_BIN /usr/src/tensorrt/bin/trtexec)
+if(NOT EXISTS TRTEXEC_DEFAULT_BIN)
+  include(FetchContent)
+  if(${TENSORRT_VERSION} VERSION_EQUAL 8.4.2)
+    set(TENSORRT_VERSION 8.4.1)
+  endif()
+  fetchcontent_declare(tensorrt
```

Comment:
[style] Seeing the official example https://cmake.org/cmake/help/latest/module/FetchContent.html, using `FetchContent` is recommended? :thinking: 

![image](https://user-images.githubusercontent.com/31987104/189508401-398b7161-3b8c-4df0-96f1-1718fb0884c0.png)

ROS 2's guideline doesn't mention that.
https://docs.ros.org/en/rolling/The-ROS2-Project/Contributing/Code-Style-Language-Versions.html#id6

Reply:
Or, as I'm trying to support language-specific dictionaries, we can add CMake function names there.
https://github.com/tier4/cspell-dicts/blob/b5286d9ac4c3aadc56de1218180e999191ac264d/cmake/words.txt#L1

Reply:
This is to avoid `Do not use mixed case commands [readability/wonkycase]` by `ament_lint_cmake`.

Reply:
I see. I think we can consider improving the rule in the future.
But for now, I'll add the functions to the dictionary.